src/ddprof_file_info.cc

```diff
@@ -5,11 +5,81 @@
 
 #include "ddprof_file_info.hpp"
 
+#include <fcntl.h>
+#include <unistd.h>
+#include "ddres_exception.hpp"
+
+#include "logger.hpp"
+
 namespace ddprof {
 
+
 bool FileInfoInodeKey::operator==(const FileInfoInodeKey &o) const {
   // default comparison is c++20
   return _inode == o._inode && _sz == o._sz;
 }
 
+FileInfoValue::FileInfoValue(FileInfo &&info, FileInfoId_t id, int fd)
+    : _info(info), _id(id) {
+  _fd = dup(fd);
```

Comment:
It makes me slightly uneasy to duplicate file descriptors knowing that the underlying file offset is shared.
Are we guaranteed that dwfl will independently of the file descriptor current offset ?

Reply:
This is a very valid concern.